controllers/builder/kogitoserverlessbuild_manager.go

```diff
@@ -54,7 +63,21 @@ func (k *sonataFlowBuildManager) GetOrCreateBuild(workflow *operatorapi.SonataFl
 			if plat, err = platform.GetActivePlatform(k.ctx, k.client, workflow.Namespace); err != nil {
 				return nil, err
 			}
-			buildInstance.Spec.BuildTemplate = plat.Spec.Build.Template
+			fmt.Println("XXXX original build template")
+			fmt.Println(plat.Spec.Build.Template)
+
+			workflowBuildTemplate := plat.Spec.Build.Template.DeepCopy()
+			if workflow.Spec.Persistence != nil && workflow.Spec.Persistence.PostgreSQL != nil {
+				//TODO WM, remove this comment
```

Comment:
@ricardozanini , here the idea is: when we create the SonataFlowBuild, we add the persistence related extensions and maven_build_args accordingly.
We do a smart adding...., if we detect that users already added stuff related to the persistence at the SonataFlowPlatform, we respect these settings.